src/celeritas/ext/detail/GeantPhysicsList.cc

```diff
@@ -340,58 +335,94 @@ void GeantPhysicsList::add_e_processes(G4ParticleDefinition* p)
         }
     }
 
+    using MMS = MscModelSelection;
+
+    // Energy limit between MSC models when multiple models are used
+    double msc_energy_limit = G4EmParameters::Instance()->MscEnergyLimit();
+    bool set_energy_limit = options_.msc == MMS::urban_wentzelvi
+                            || options_.msc == MMS::gs_wentzelvi;
+
     if (options_.coulomb_scattering)
     {
         // Coulomb scattering: G4eCoulombScatteringModel
-        double msc_energy_limit = G4EmParameters::Instance()->MscEnergyLimit();
-
-        auto process = std::make_unique<G4CoulombScattering>();
-        auto model = std::make_unique<G4eCoulombScatteringModel>();
-        process->SetMinKinEnergy(msc_energy_limit);
-        model->SetLowEnergyLimit(msc_energy_limit);
-        model->SetActivationLowEnergyLimit(msc_energy_limit);
-        process->SetEmModel(model.release());
-        physics_list->RegisterProcess(process.release(), p);
+        if (options_.msc == MMS::urban)
```

Comment:
Maybe we should decouple single and multiple scattering for registering processes. Our `GeantPhysicsOptions` has `coulomb_scattering` and `msc` as separate fields, so I feel like it also makes sense to have something like

```cpp
if (options_.coulomb_scattering)
{
    // Enable Coulomb Scattering
}
if (options_.msc)
{
    // Register physics according to selection and issue warnings for energy ranges as needed
}
```

Reply:
I think if we have both Coulomb and Wentzel VI enabled we should be using the same energy range for both, and if if only single Coulomb scattering is enabled it should cover the entire energy range. I'm not sure it makes sense to have Coulomb scattering enabled in the same energy range as either Urban MSC or GS (although I suppose we could allow it and issue a warning?). Maybe @whokion can comment?

Reply:
What @amandalund  propose is consistent with practical user cases and the current Geant4 default physics list support.  In principle, SS (exclusively without any other other msc model) in all energy ranges is the most accurate  Coulomb scattering model, but may not be used in any LHC-like HEP production as it is computational very expensive (due to the high cross section in typical HEP calorimeters). So, it is mostly used for the purpose of model developments (i.e., to benchmark accuracy of other MSC models or validation) or special studies for detector design or optimization including back scatterings - one of exceptional examples of using SS alone in all range is in a sparse or low density medium (such as gaseous material).  The reason that SS is combined in the same energy range of the high energy MSC model is a kind of trade-off of adding the more accuracy with an affordable additional computing cost as the SS in the low energy is very prohibitively expensive - of course, the question whether the same energy cutoff (100MeV) with the high energy (WentzelVI) MSC model should be used or not is a practical choice for conveniences for the simpler setup for physics processes or lists.   As a summary, it will be good to support 1) [Urbarn]+[[WentzelVI]+[SS]] (Geant4 default), 2) [Urban] + [WentzelVI] (a.k.a. CMS EMM), 3) [SS] in the all range - of course [Urbarn] can be substituted by [GS] if GS is implemented and the more accurate simulation is needed for the additional cost of computing time.

Reply:
Thanks for the detailed answer @whokion!